Sources/SWBCore/Settings/Settings.swift

```diff
@@ -2304,10 +2304,15 @@ private class SettingsBuilder {
         // subtypes in all cases (x86_64h, arm64e, etc.). Also, NXGetLocalArchInfo is not guaranteed to include the CPU_ARCH_ABI64 mask in its returned cputype
         // (for example on a non-Haswell CPU we get a general x86 CPU subtype which won't infer 64-bit like Haswell does), so we manually add the mask if the CPU is 64-bit capable.
         let fallbackArch = "undefined_arch"
+        #if canImport(Darwin)
```

Comment:
Please use:

```swift
core.hostOperatingSystem == .macOS
```

instead of idefs, to maximize compilation coverage across all branches.